pkg/processor/trigger/trigger.go

```diff
@@ -424,7 +437,114 @@ func (at *AbstractTrigger) prepareEvent(event nuclio.Event, workerInstance *work
 	}
 
 	// Not a cloud event
-	event.SetID(nuclio.ID(uuid.New().String()))
+	// id won't be empty only in batching mode
```

Comment:
what does that mean?

Reply:
@TomerShor this code is for BC, because in batching mode we set event id much earlier than here, but for normal flow we set it here. So, in order to avoid changing of id when batching is enabled in that place, I've added this trick.

Reply:
Got it 👍 
I would change it to something like (instead of both comments):
`// In batch-mode and cloud event an ID already exists at this point, so don't override it`